pkg/controller/utils/component.go

```diff
@@ -301,6 +301,8 @@ func ensureOSSchedulingRestrictions(obj client.Object, osType render.OSType) {
 
 	var podSpecs []*v1.PodSpec
 	switch obj.(type) {
+	case *v1.PodTemplate:
```

Comment:
I'm a little confused, how does a PodTemplate end up getting here, these should be renderable items, is a pod template being passed as a resource that should be rendered?

Reply:
Nm @robbrockbank answer this and it's a valid type that you can apply.